lwt/nbd_lwt_unix.ml

```diff
@@ -13,13 +13,76 @@
  *)
 open Nbd
 open Channel
-open Lwt
 
-let of_fd fd =
-  let read = Lwt_cstruct.(complete (read fd)) in
-  let write = Lwt_cstruct.(complete (write fd)) in
-  let close () = Lwt_unix.close fd in
-  { read; write; close }
+let return = Lwt.return
+let (>>=) = Lwt.(>>=)
+
+type tls_role =
+  | TlsClient of Ssl.context
+  | TlsServer of Ssl.context
+
+let debug_io = ref false
+
+(* TODO move this function into a library used by this and other executables *)
+(* Also the read/write/close functions that are inside tls_channel_of_fd *)
+let io_complete name offset op fd buffer =
```

Comment:
We should definitely fix the source as well where this code came from. This is why we should have only one implementation.